### DISCUSSION TITLE
A clarification on Pybricks…

### DISCUSSION THREAD

**Discussion by gianlucadegiorgio** [pybricks/support#1036] [General]
Good morning everyone. Recently my son is using Robot Inventor. I noticed that it is possible to create a project directly in Python from the Lego App. I would like to understand if there is and what is the difference between this Python programming and what could be done with the Pybricks firmware

**Comment by laurensvalk** [comment]
Here are just a few differences:

- Code starts and runs much faster.
- You can split your code into modules, and import code between them. This lets you organize your work.
- Auto completion of code as you type.
- Includes detailed documentation.
- Pybricks runs on all LEGO hubs, not just SPIKE/MINDSTORMS.
- All motors/sensors are supported, including some third party devices.
- Includes support for remote control. Can be useful to test your mechanical builds with less code.
- Includes interactive input/output window terminal.
- Consistent platform with user API that hasn't had breaking changes since it was first released.
